packages/website/src/components/config/ConfigEslint.tsx

```diff
@@ -1,90 +1,60 @@
+  if (schemaFile) {
+    const schema = JSON.parse(schemaFile) as JSONSchema4;
+    if (schema.type === 'object' && schema.properties?.rules?.properties) {
+      return schemaToConfigOptions(
+        schema.properties.rules.properties,
+      ).reverse();
+    }
   }
-  return false;
-}
 
-function checkOptions(rule: [string, unknown]): rule is [string, RuleEntry] {
-  if (Array.isArray(rule[1])) {
-    return rule[1].length > 0 && checkSeverity(rule[1][0]);
-  }
-  return checkSeverity(rule[1]);
+  return [];
 }
 
-function ConfigEslint(props: ConfigEslintProps): JSX.Element {
-  const { isOpen, config, onClose: onCloseProps, ruleOptions } = props;
+function ConfigEslint({
+  isOpen,
+  onClose: onCloseProps,
+  system,
+}: ConfigEslintProps): JSX.Element {
   const [options, updateOptions] = useState<ConfigOptionsType[]>([]);
-  const [configObject, updateConfigObject] = useState<EslintRC>();
+  const [configObject, updateConfigObject] = useState<TSESLint.Linter.Config>();
 
   useEffect(() => {
     if (isOpen) {
+      updateOptions(readConfigSchema(system));
+      const config = system.readFile('/.eslintrc');
       updateConfigObject(parseESLintRC(config));
     }
-  }, [isOpen, config]);
-
-  useEffect(() => {
-    updateOptions([
-      {
-        heading: 'Rules',
-        fields: ruleOptions
-          .filter(item => item.name.startsWith('@typescript'))
-          .map(item => ({
-            key: item.name,
-            label: item.description,
-            type: 'boolean',
-            defaults: ['error', 2, 'warn', 1, ['error'], ['warn'], [2], [1]],
-          })),
-      },
-      {
-        heading: 'Core rules',
-        fields: ruleOptions
-          .filter(item => !item.name.startsWith('@typescript'))
-          .map(item => ({
-            key: item.name,
-            label: item.description,
-            type: 'boolean',
-            defaults: ['error', 2, 'warn', 1, ['error'], ['warn'], [2], [1]],
-          })),
-      },
-    ]);
-  }, [ruleOptions]);
+  }, [isOpen, system]);
 
   const onClose = useCallback(
-    (newConfig: Record<string, unknown>) => {
-      const cfg = Object.fromEntries(
-        Object.entries(newConfig)
-          .map<[string, unknown]>(([name, value]) =>
-            Array.isArray(value) && value.length === 1
-              ? [name, value[0]]
-              : [name, value],
-          )
-          .filter(checkOptions),
-      );
-      if (!shallowEqual(cfg, configObject?.rules)) {
-        onCloseProps({
-          eslintrc: toJson({ ...(configObject ?? {}), rules: cfg }),
-        });
-      } else {
-        onCloseProps();
+    (newRules: Record<string, unknown>) => {
+      if (!shallowEqual(newRules, configObject?.rules)) {
+        system.writeFile(
+          '/.eslintrc',
+          toJson({ ...(configObject ?? {}), rules: newRules }),
+        );
       }
+      onCloseProps(false);
     },
-    [onCloseProps, configObject],
+    [onCloseProps, configObject, system],
   );
```

Comment:
This `onClose` pattern (here, `ConfigTypeScript.tsx`) means the configs never get "flushed" into the system & factored into linting / type checking until the tab is closed, right?

If so: can I suggest having them come into play in (perhaps debounced) real time? Eventually it'd be nice to have the playground be able to visualize multiple points of interest at a time. That could be showing all three files side-by-side, and/or showing both the ESLint and TS outputs at the same time.

I think doing so would mean the `if (isOpen)`s in this & the other `Config*.tsx` components would be unnecessary. The components would all effectively be "open" in their interpretation at all times, potentially actively writing to the system.

Reply:
that was my general idea, to make it more universal and just use schema to generate visual editor, i will try to improve this in feature